segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -438,20 +439,24 @@ public void createSegment(CreateSegment createStreamSegment) {
         Timer timer = new Timer();
         final String operation = "createSegment";
 
+        Preconditions.checkArgument(createStreamSegment.getRolloverSizeBytes() >= 0,
```

Comment:
I don't recommend we do it this way. At this point there is absolutely nothing that anyone can do to fix this. The Client has already sent its request to the Controller and got an ack back, and at this point the Controller is just going to retry this indefinitely, which will obviously lead nowhere if it keeps passing a bad value.

Let's log this as a `WARN` and interpret any invalid values as `0`.